mesonbuild/mesonlib/vsenv.py

```diff
@@ -67,8 +66,10 @@ def _setup_vsenv(force: bool) -> bool:
         # VS installer instelled but not VS itself maybe?
         raise MesonException('Could not parse vswhere.exe output')
     bat_root = pathlib.Path(bat_info[0]['installationPath'])
-    if platform.machine() == 'ARM64':
-        bat_path = bat_root / 'VC/Auxiliary/Build/vcvarsx86_arm64.bat'
+    if windows_detect_native_arch().lower() == 'arm64':
```

Comment:
Why do we need .lower() here? Doesn't the function always return a lowercase string anyway?

Reply:
fixed